index.bs

```diff
@@ -6208,6 +6080,474 @@ The following abstract operations are a grab-bag of utilities.
+example, instead of testing if the stream is [=ReadableStream/locked=], attempt to
+[=ReadableStream/get a reader=] and handle any exception.)
+
+<p algorithm>A {{ReadableStream}} |stream| is <dfn export for="ReadableStream">readable</dfn> if
+|stream|.[=ReadableStream/[[state]]=] is "`readable`".
+
+<p algorithm>A {{ReadableStream}} |stream| is <dfn export for="ReadableStream">closed</dfn> if
+|stream|.[=ReadableStream/[[state]]=] is "`closed`".
+
+<p algorithm>A {{ReadableStream}} |stream| is <dfn export for="ReadableStream">errored</dfn> if
+|stream|.[=ReadableStream/[[state]]=] is "`errored`".
+
+<p algorithm="ReadableStream locked">A {{ReadableStream}} |stream| is <dfn export
+for="ReadableStream">locked</dfn> if ! [$IsReadableStreamLocked$](|stream|) returns true.
+
+<div algorithm>
+ <p>A {{ReadableStream}} |stream| is <dfn export for="ReadableStream"
+ id="is-readable-stream-disturbed">disturbed</dfn> if |stream|.[=ReadableStream/[[disturbed]]=] is
+ true.
+
+ <p class="note">This indicates whether the stream has ever been read from or canceled. Even more so
+ than other predicates in this section, it is best consulted sparingly, since this is not
+ information web developers have access to even indirectly. As such, branching platform behavior on
+ it is undesirable.
+</div>
+
+<h3 id="other-specs-ws">Writable streams</h3>
+
+<h4 id="other-specs-ws-creation">Creation and manipulation</h4>
+
+<div algorithm="create a WritableStream">
+ To <dfn export for="WritableStream" lt="create|creating">create</dfn> a {{WritableStream}} given an
+ algorithm <dfn export for="WritableStream/create"><var>writeAlgorithm</var></dfn>, an optional
+ algorithm <dfn export for="WritableStream/create"><var>closeAlgorithm</var></dfn>, an optional
+ algorithm <dfn export for="WritableStream/create"><var>abortAlgorithm</var></dfn>, an optional
+ number <dfn export for="WritableStream/create"><var>highWaterMark</var></dfn> (default 1), an
+ optional algorithm <dfn export for="WritableStream/create"><var>sizeAlgorithm</var></dfn>, perform
+ the following steps. |writeAlgorithm| must be an algorithm that accepts a [=chunk=] object and
+ returns a promise. If given, |closeAlgorithm| and |abortAlgorithm| must return a promise. If
+ given, |sizeAlgorithm| must be an algorithm accepting [=chunk=] objects and
+ returning a number; and if given, |highWaterMark| must be a non-negative, non-NaN number.
+
+ 1. Let |startAlgorithm| be an algorithm that returns undefined.
+ 1. Let |closeAlgorithmWrapper| be an algorithm that runs these steps:
+  1. If |closeAlgorithm| was given, return the result of running it.
+  1. Return [=a promise resolved with=] undefined.
+ 1. Let |abortAlgorithmWrapper| be an algorithm that runs these steps:
+  1. If |abortAlgorithm| was given, return the result of running it.
+  1. Return [=a promise resolved with=] undefined.
+ 1. If |sizeAlgorithm| is not given, then set it to an algorithm that returns 1.
```

Comment:
Why is "is not given" present tense when in the previous steps "was given" is past tense?

Reply:
I suspect this is because the steps in *closeAlgorithmWrapper* and *abortAlgorithmWrapper* are run **after** the main steps complete.

Reply:
That makes sense.

Reply:
Consistency would be nicer; I'll stick with "was". It looks like this is a preexisting problem so it'll touch a few algorithms.